### PR TITLE
ga4gh

### PR DESCRIPTION
**Pull request by BaranekD**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code or performance improvement
- [ ] Documentation changes or improvements

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My code has tests
- [x] My code has been tested manually
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/cz/muni/ics/perunproxyapi/persistence/models/PassportVisa.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        return "PassportVisa{" +
+//					"jwt='" + jwt + '\'' +
+                "  type=" + type +
+                ", sub=" + sub +
+                ", iss=" + iss +
+                ", value=" + value +
+                ", verified=" + verified +
+                ", linkedIdentity=" + linkedIdentity +
+                '}';
+    }
```
Comment (dBucik) [inline src/main/java/cz/muni/ics/perunproxyapi/persistence/models/PassportVisa.java:66]: I think we can use lomboks annotation and specify the JWT field to be excluded

File: src/main/java/cz/muni/ics/perunproxyapi/presentation/rest/controllers/ProxyUserProtectedController.java [hunk 2]
```diff
@@ -190,4 +196,10 @@ public UserDTO findByPerunUserId(@RequestParam(value = USER_ID) Long userId)
         return facade.getAllEntitlements(login);
     }
 
+    @ResponseBody
+    @GetMapping(value = "/ga4gh", produces = APPLICATION_JSON_VALUE)
+    public JsonNode ga4gh(@RequestParam(value = USER_ID) Long userId) throws NoSuchAlgorithmException, MalformedURLException, PerunConnectionException, FileNotFoundException, URISyntaxException, PerunUnknownException, InvalidKeySpecException {
```
Comment (dBucik) [inline src/main/java/cz/muni/ics/perunproxyapi/presentation/rest/controllers/ProxyUserProtectedController.java:201]: You are missing validation for the USER_ID param.

File: src/main/java/cz/muni/ics/perunproxyapi/presentation/rest/interceptors/AddHeaderInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * @author Martin Kuba <makub@ics.muni.cz>
+ */
+public class AddHeaderInterceptor implements ClientHttpRequestInterceptor {
+
+    private final String header;
+    private final String value;
+
+    public AddHeaderInterceptor(String header, String value) {
+        this.header = header;
```
Comment (dBucik) [inline src/main/java/cz/muni/ics/perunproxyapi/presentation/rest/interceptors/AddHeaderInterceptor.java:22]: Header should not be empty string :)

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -53,3 +53,8 @@ management:
       enabled: false
     build:
       enabled: false
+
+ga4gh:
+  path: /etc/perun/keystore.jwks
```
Comment (dBucik) [inline src/main/resources/application.yml:58]: I would rename this field to "config_file_path" or something similar that expresses more what value should the field contain. Even better would be something like "jwk_keystore_file_path"

File: src/test/resources/application.yml [hunk 5]
```diff
@@ -47,3 +47,8 @@ facade:
 security:
   basicauth:
     path: "${PWD}/src/test/resources/configs/api-auth.yml"
+
+ga4gh:
+  path: /etc/perun/keystore.jwks
```
Comment (dBucik) [inline src/test/resources/application.yml:52]: Reflect changes made in the other application.yml file here as well.

**Review by dBucik — CHANGES_REQUESTED** [review]

**Comment by vyskocilpavel** [comment]
@BaranekD Please rebase this PR. Thanks

**Pull request merged**